src/main/scala/edu/ie3/util/geo/GeoUtils.scala

```diff
@@ -0,0 +1,341 @@
+    * and return a new, adapted instance of [[Point]]
+    *
+    * @param p1
+    *   the point that should be adapted
+    * @return
+    *   the adapted point with a slightly changed coordinate
+    */
+  private def buildSafePoint(p1: Point): Point = {
+    val safeCoord = buildSafeCoord(p1.getCoordinate)
+    val safeCoordSeq = new CoordinateArraySequence(Array[Coordinate](safeCoord))
+    new Point(safeCoordSeq, p1.getFactory)
+  }
+
+  /** Adapted [[Coordinate]] x, [[Coordinate]] y and [[Coordinate]] z of the
+    * provided [[Coordinate]] by 1e-13 and return a new, adapted instance of
+    * [[Coordinate]]
+    *
+    * @param coord
+    *   the coordinate that should be adapted
+    * @return
+    *   the adapted coordinate with slightly changed x,y,z values
+    */
+  private def buildSafeCoord(coord: Coordinate): Coordinate = {
+    val modVal: Double = 1e-13
+    val p1X: Double = coord.getX + modVal
+    val p1Y: Double = coord.getY + modVal
+    val p1Z: Double = coord.getZ + modVal
+    new Coordinate(p1X, p1Y, p1Z)
+  }
+
+  /** Calculates and orders the coordinate distances from a base coordinate to a
+    * list of coordinates
+    *
+    * @param baseCoordinate
+    *   the base coordinate
+    * @param coordinates
+    *   the coordinates to calculate the distance for
+    * @return
+    *   a sorted set of [[CoordinateDistance]]
+    */
+  def calcOrderedCoordinateDistances(
+      baseCoordinate: Point,
+      coordinates: Array[Point]
+  ): SortedSet[CoordinateDistance] = {
+    val coordinateDistances = coordinates.map(coordinate =>
+      new CoordinateDistance(baseCoordinate, coordinate)
+    )
+    TreeSet(coordinateDistances: _*)
+  }
+
+  /** Calculates the great circle distance between two coordinates
+    *
+    * @param latA
+    *   latitude of coordinate a
+    * @param longA
+    *   longitude of coordinate a
+    * @param latB
+    *   latitude of coordinate b
+    * @param longB
+    *   longitude of coordinate b
+    * @return
+    *   the distance between the coordinates as a quantity
+    */
+  def calcHaversine(
+      latA: Double,
+      longA: Double,
+      latB: Double,
+      longB: Double
+  ): ComparableQuantity[Length] = {
+    val r = EARTH_RADIUS
+    val dLat = Quantities.getQuantity(toRadians(latB - latA), RADIAN)
+    val dLon = Quantities.getQuantity(toRadians(longB - longA), RADIAN)
+    val a = pow(sin(dLat.getValue.doubleValue / 2), 2) + cos(
+      toRadians(latA)
+    ) * cos(toRadians(latB)) * pow(sin(dLon.getValue.doubleValue / 2), 2)
+    val c = 2 * atan2(sqrt(a), sqrt(1 - a))
+    r.multiply(c)
+  }
+
+  /** Builds a convex hull from a set of latitude/longitude coordinates.
+    *
+    * @param coordinates
+    *   the coordinates to consider
+    * @return
+    *   a Try of the resulting polygon
+    */
+  def buildConvexHull(coordinates: Set[Coordinate]): Try[Polygon] = {
+    val projectedCoordinates = coordinates
```

Comment:
What is the reason to reassign the coordinates here?

Reply:
remnent of a previous implementation